### PR TITLE
treewide: switch to fetchFromGitHub

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
use `fetchFromGitHub`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/libcangjie/default.nix [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 
 stdenv.mkDerivation rec {
   pname = "libcangjie";
   version = "1.4_rev_${rev}";
   rev = "a73c1d8783f7b6526fd9b2cc44a669ffa5518d3d";
 
-  src = fetchurl {
-    url = "https://github.com/Cangjians/libcangjie/archive/${rev}.tar.gz";
-    sha256 = "0i5svvcx099fc9hh5dvr3gpb1041v6vn5fnylxy82zjy239114lg";
+  src = fetchFromGitHub {
```
Comment (prusnak) [inline pkgs/development/libraries/libcangjie/default.nix:8]: project moved to https://gitlab.freedesktop.org/cangjie/libcangjie

**Review by prusnak — CHANGES_REQUESTED** [review]

**Review by prusnak — APPROVED** [review]

**Comment by Stunkymonkey** [comment]
what does the failed check mean?

**Review by dasj19 — APPROVED** [review]

**Comment by prusnak** [comment]
> what does the failed check mean?

It seems you've got the hash for `libcec` wrong?

**Comment by Stunkymonkey** [comment]
Result of `nixpkgs-review pr 153482` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>grass</li>
  </ul>
</details>
<details>
  <summary>36 packages built:</summary>
  <ul>
    <li>LASzip2</li>
    <li>cloudcompare</li>
    <li>fehlstart</li>
    <li>graphene-hardened-malloc</li>
    <li>keybinder</li>
    <li>kodi (kodiPackages.kodi)</li>
    <li>kodi-gbm</li>
    <li>kodi-wayland</li>
    <li>kodiPackages.a4ksubtitles</li>
    <li>kodiPackages.archive_tool</li>
    <li>kodiPackages.iagl</li>
    <li>kodiPackages.inputstream-adaptive</li>
    <li>kodiPackages.inputstream-ffmpegdirect</li>
    <li>kodiPackages.inputstream-rtmp</li>
    <li>kodiPackages.joystick</li>
    <li>kodiPackages.kodi-platform</li>
    <li>kodiPackages.libretro</li>
    <li>kodiPackages.libretro-genplus</li>
    <li>kodiPackages.libretro-mgba</li>
    <li>kodiPackages.libretro-snes9x</li>
    <li>kodiPackages.netflix</li>
    <li>kodiPackages.pvr-hdhomerun</li>
    <li>kodiPackages.pvr-hts</li>
    <li>kodiPackages.pvr-iptvsimple</li>
    <li>kodiPackages.steam-controller</li>
    <li>kodiPackages.vfs-libarchive</li>
    <li>kodiPackages.vfs-sftp</li>
    <li>libLAS</li>
    <li>libcangjie</li>
    <li>libcec</li>
    <li>lxpanel</li>
    <li>moonlight-embedded</li>
    <li>python38Packages.pycangjie</li>
    <li>python38Packages.pycec</li>
    <li>python39Packages.pycangjie</li>
    <li>python39Packages.pycec</li>
  </ul>
</details>

**Comment by Stunkymonkey** [comment]
compiles on my machine.

File: pkgs/development/libraries/libcec/default.nix [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
   version = "6.0.2";
 
-  src = fetchurl {
-    url = "https://github.com/Pulse-Eight/libcec/archive/libcec-${version}.tar.gz";
-    sha256 = "0xrkrcgfgr5r8r0854bw3i9jbq4jmf8nzc5vrrx2sxzvlkbrc1h9";
+  src = fetchFromGitHub {
+    owner = "Pulse-Eight";
+    repo = "libcec";
+    rev = "libcec-${version}";
+    sha256 = "sha256-txHJxKAyk7e9fNQvVkLnlnnQ7EJTB58enDe8rgy99fo=";
```
Comment (mohe2015) [inline pkgs/development/libraries/libcec/default.nix:11]:
```suggestion
    sha256 = "sha256-OWqCn7Z0KG8sLlfMWd0btJIFJs79ET3Y1AV/y/Kj2TU=";
```

**Comment by mohe2015** [comment]
@Stunkymonkey you probably have the other hash in your local nix store and therefore this doesn't get detected

**Comment by Stunkymonkey** [comment]
thanks @mohe2015 :+1: . It works now!

**Pull request merged**